internal/impl/redpanda/migrator/migrator.go

```diff
@@ -0,0 +1,535 @@
+
+Metrics:
+
+The component exposes comprehensive metrics for monitoring migration operations:
+
+Topic Migration Metrics:
+- `+"`migrator_topics_created_total`"+` (counter): Total number of topics successfully created on the destination cluster
+- `+"`migrator_topic_create_errors_total`"+` (counter): Total number of errors encountered when creating topics
+- `+"`migrator_topic_create_latency_ns`"+` (timer): Latency in nanoseconds for topic creation operations
+
+Schema Registry Migration Metrics:
+- `+"`migrator_sr_schemas_created_total`"+` (counter): Total number of schemas successfully created in the destination schema registry
+- `+"`migrator_sr_schema_create_errors_total`"+` (counter): Total number of errors encountered when creating schemas
+- `+"`migrator_sr_schema_create_latency_ns`"+` (timer): Latency in nanoseconds for schema creation operations
+- `+"`migrator_sr_compatibility_updates_total`"+` (counter): Total number of compatibility level updates applied to subjects
+- `+"`migrator_sr_compatibility_update_errors_total`"+` (counter): Total number of errors encountered when updating compatibility levels
+- `+"`migrator_sr_compatibility_update_latency_ns`"+` (timer): Latency in nanoseconds for compatibility level update operations
+
+Consumer Group Migration Metrics (with group label):
+- `+"`migrator_cg_offsets_translated_total`"+` (counter): Total number of offsets successfully translated per consumer group
+- `+"`migrator_cg_offset_translation_errors_total`"+` (counter): Total number of errors encountered when translating offsets per consumer group
+- `+"`migrator_cg_offset_translation_latency_ns`"+` (timer): Latency in nanoseconds for offset translation operations per consumer group
+- `+"`migrator_cg_offsets_committed_total`"+` (counter): Total number of offsets successfully committed per consumer group
+- `+"`migrator_cg_offset_commit_errors_total`"+` (counter): Total number of errors encountered when committing offsets per consumer group
+- `+"`migrator_cg_offset_commit_latency_ns`"+` (timer): Latency in nanoseconds for offset commit operations per consumer group
+
+This component must be paired with the `+"`redpanda_migrator`"+` input in the same pipeline.`).
+		Example(
+			"Basic migration",
+			"Migrate topics, schemas and consumer groups from source to destination.",
+			`input:
+  redpanda_migrator:
+    seed_brokers: ["source:9092"]
+    topics: ["orders", "payments"]
+    consumer_group: "migration"
+
+output:
+  redpanda_migrator:
+    seed_brokers: ["destination:9092"]
+    # Write to the same topic name
+    topic: ${! metadata("kafka_topic") }
+    schema_registry:
+      url: "http://dest-registry:8081"
+      translate_ids: true
+    consumer_groups:
+      interval: 1m
+`).
+		// Kafka fields
+		Fields(kafka.FranzConnectionFields()...).
+		Fields(kafka.FranzProducerFields()...).
+		Fields(kafka.FranzWriterConfigFields()...).
```

Comment:
One other thing I noticed is that by default we should be preserving the key, partition, and topic. We should not require everyone setup manual partitioning and all the other things that they expect from a migration. We need to tweak the defaults for the FranzWriter here for this. I'm not sure how we want to do it, but in your above example they should preserve the key, the partition and topic by default, and be able to override the topic for example if they want to remap things.

Reply:
@rockwotj please note [0f080d4](https://github.com/redpanda-data/connect/pull/3564/commits/0f080d426ce822faec4fd4b47deb812e9eb109cd)